lib/Doctrine/DBAL/Portability/Statement.php

```diff
@@ -213,6 +213,55 @@ protected function fixRow($row, $iterateRow, $fixCase)
     }
 
     /**
+     * Fixes column values according to portability settings
+     *
+     * @param mixed[] $values Column values indexed by row number
+     * @param bool $emptyToNull Should empty strings be converted to NULL
+     * @param bool $rTrim Should strings be right-trimmed
+     * @return mixed[]
+     */
+    private function fixColumn(array $values, $emptyToNull, $rTrim)
+    {
+        if (!$emptyToNull && !$rTrim) {
```

Comment:
Another idea that comes to mind is to normalize the rows at the beginning if `PDO::FETCH_COLUMN` is given and denormalize again at the end? Like expand the rows to a multi-dimensional array at the beginning and flatten again after portability adjustments? Not sure if that might be a better solution.